code/game/objects/items.dm

```diff
@@ -187,6 +187,103 @@ var/global/image/fire_overlay = image("icon" = 'icons/effects/fire.dmi', "icon_s
 		msg += "*--------*"
 		user << msg
 
+	if(user.robustness_scanner)
+		user << "[list_obj_robustness()]"
+
+/obj/item/proc/list_obj_robustness()
+	var/nil = "<b>Null</b>"
+	var/vlow = "<font color='red'>Very low</font>"
+	var/low = "<font color='red'>Low</font>"
+	var/med = "<font color='#FF8000'>Medium</font>"
+	var/high = "<font color='blue'>High</font>"
+	var/vhigh = "<font color='green'>Very high</font>"
+
+	var/msg = "*--------* <BR>"
+
+	if(force || throwforce)
+		msg += "Weapon potential:<BR>"
+		var/force_str
+		var/throwforce_str
+
+		switch(force)
+			if(0)
+				force_str += "[nil]"
+			if(1 to 5)
+				force_str += "[vlow]"
+			if(6 to 10)
+				force_str += "[low]"
+			if(11 to 15)
+				force_str += "[med]"
+			if(16 to 20)
+				force_str += "[high]"
+			if(21 to INFINITY)
+				force_str += "[vhigh]"
+
+		switch(throwforce)
+			if(0)
+				throwforce_str = "[nil]"
+			if(1 to 5)
+				throwforce_str = "[vlow]"
+			if(6 to 10)
+				throwforce_str = "[low]"
+			if(11 to 15)
+				throwforce_str = "[med]"
+			if(16 to 20)
+				throwforce_str = "[high]"
+			if(21 to INFINITY)
+				throwforce_str = "[vhigh]"
+
+		msg += "[force_str] melee damage in its current state. [throwforce_str] throwing damage in its current state."
+
+	if(istype(src, /obj/item/clothing))
+		var/damdesc
+		var/damquality
+
+		msg += "Armor:<BR>"
+
+		for(var/damtype in armor)
+
+			switch(armor["[damtype]"])
+				if(0)
+					break

```

Comment:
Should be continue? If the first armour type is 0 but second isnt you won't see second..